File: src/test/java/no/nav/syfo/TestApplicationConfig.java
package no.nav.syfo;

import no.nav.syfo.services.LdapService;
import org.springframework.context.annotation.*;

import static org.mockito.Mockito.mock;

@Configuration
@Profile({"test"})
@Import(LocalApplicationConfig.class)
public class TestApplicationConfig {

    @Bean
    @Primary
    public LdapService ldapServiceMock() {
        return mock(LdapService.class);
    }

}

File: src/test/kotlin/no/nav/syfo/config/TestApplicationConfig.kt
package no.nav.syfo.config

import no.nav.syfo.LocalApplicationConfig
import no.nav.syfo.services.LdapService
import org.mockito.Mockito
import org.springframework.context.annotation.*

@Configuration
@Profile("test")
@Import(LocalApplicationConfig::class)
class TestApplicationConfig {
    @Bean
    @Primary
    fun ldapServiceMock(): LdapService {
        return Mockito.mock(LdapService::class.java)
    }
}
